File: Writer_App/lib/Model/WriterWorkModel.dart
class Writerworkmodel {

  int ?userId;
  int? UserOrderId;
  String? fileUrl;
  String? fileName;
  int? fileSize;
  int? filePageCount;
  int? writerId;
  String? cancellationReason;
  DateTime? orderCreatedAt;
  DateTime? orderCompletedAt;
  String? writerAssignmentStatus;
  String? writerSuggestionText;

  Writerworkmodel({
    this.userId,
    this.UserOrderId,
    this.fileUrl,
    this.fileName,
    this.fileSize,
    this.filePageCount,
    this.writerId,
    this.cancellationReason,
    this.orderCreatedAt,
    this.orderCompletedAt,
    this.writerAssignmentStatus,
    this.writerSuggestionText,
  });

  factory Writerworkmodel.fromJson(Map<String, dynamic> json) {

    return Writerworkmodel(
      userId: json['userId'] as int?,
      UserOrderId: json['UserOrderId'] is int
          ? json['UserOrderId']
          : int.tryParse(json['UserOrderId']?.toString() ?? ''),

      fileUrl: json['fileUrl'] as String?,

      fileName: json['fileName'] as String?,
      fileSize: json['fileSize'] as int?,
      filePageCount: json['filePageCount'] as int?,
      writerId: json['writerId'] as int?,
      cancellationReason: json['cancellationReason'] as String?,

      orderCreatedAt: json['orderCreatedAt'] != null
          ? DateTime.tryParse(json['orderCreatedAt'])
          : null,

      orderCompletedAt: json['orderCompletedAt'] != null
          ? DateTime.tryParse(json['orderCompletedAt'])
          : null,

      writerAssignmentStatus: json['writerAssignmentStatus'] as String?,
      writerSuggestionText: json['writerSuggestionText'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'UserOrderId': UserOrderId,
      'fileUrl': fileUrl,
      'fileName': fileName,
      'fileSize': fileSize,
      'filePageCount': filePageCount,
      'writerId': writerId,
      'cancellationReason': cancellationReason,
      'orderCreatedAt': orderCreatedAt?.toIso8601String(),
      'orderCompletedAt': orderCompletedAt?.toIso8601String(),
      'writerAssignmentStatus': writerAssignmentStatus,
      'writerSuggestionText': writerSuggestionText,

    };
  }
}

File: Backend/src/main/java/com/example/likhwao/Model/WriterWorkModel.java
package com.example.likhwao.Model;

import java.sql.Date;

public class WriterWorkModel {
    

    public  Long userId;
    public  Long UserOrderId;

  public String fileUrl;
 public  String fileName;
 public  Long fileSize;
 public int filePageCount;
 public Long writerId;
  public String cancellationReason;

 public  Date orderCreatedAt;
 public  Date orderCompletedAt;
  public String  writerAssignmentStatus;
  public String writerSuggestionText;

  public Long getUserOrderId() {
    return UserOrderId;
  }
  public void setUserOrderId(Long userOrderId) {
    UserOrderId = userOrderId;
  }
  public String getFileUrl() {
    return fileUrl;
  }
  public void setFileUrl(String fileUrl) {
    this.fileUrl = fileUrl;
  }
  public String getFileName() {
    return fileName;
  }
  public void setFileName(String fileName) {
    this.fileName = fileName;
  }
  public Long getFileSize() {
    return fileSize;
  }
  public void setFileSize(Long fileSize) {
    this.fileSize = fileSize;
  }
  public int getFilePageCount() {
    return filePageCount;
  }
  public void setFilePageCount(int filePageCount) {
    this.filePageCount = filePageCount;
  }
  public Long getWriterId() {
    return writerId;
  }
  public void setWriterId(Long writerId) {
    this.writerId = writerId;
  }
  public String getCancellationReason() {
    return cancellationReason;
  }
  public void setCancellationReason(String cancellationReason) {
    this.cancellationReason = cancellationReason;
  }
  public Date getOrderCreatedAt() {
    return orderCreatedAt;
  }
  public void setOrderCreatedAt(Date orderCreatedAt) {
    this.orderCreatedAt = orderCreatedAt;
  }
  public Date getOrderCompletedAt() {
    return orderCompletedAt;
  }
  public void setOrderCompletedAt(Date orderCompletedAt) {
    this.orderCompletedAt = orderCompletedAt;
  }
  public String getWriterAssignmentStatus() {
    return writerAssignmentStatus;
  }
  public void setWriterAssignmentStatus(String writerAssignmentStatus) {
    this.writerAssignmentStatus = writerAssignmentStatus;
  }
  public WriterWorkModel() {
  }
  public String getWriterSuggestionText() {
    return writerSuggestionText;
  }
  public void setWriterSuggestionText(String writerSuggestionText) {
    this.writerSuggestionText = writerSuggestionText;
  }
  

  public Long getUserId() {
    return userId;
  }
  public void setUserId(Long userId) {
    this.userId = userId;
  }
 

  
}
